overlays/bootstrap.nix

```diff
@@ -313,7 +315,30 @@ in {
         })));
 
     ghc = final.haskell-nix.compiler.ghc865;
-    inherit (final.buildPackages.haskell-nix.bootstrap.packages) cabal-install alex happy;
+    cabal-install =
+      # Make sure that `cabal-install` used building `cabal-install`
+      # always has `checkMaterialization = false` to avoid infinite
+      # recursion.
```

Comment:
This could do with some expansion: I suspect figuring out why this was a problem took you some time, worth recording so future people don't have to do that!